kalite/distributed/static/js/distributed/topics/views.js

```diff
@@ -56,24 +56,46 @@ var ContentAreaView = BaseView.extend({
         this.$(".content").html("").append(view.$el);
     },
 
+    should_show_rating: function() {
+        /*
+        This function determines whether a rating should be shown for the content item.
+        returns: true or false
+        */
+        var entry_available = (typeof this.model !== "undefined") && !!this.model.get("available");
+        var logged_in = typeof window.statusModel.get("user_id") !== "undefined";
+        return logged_in && entry_available;
+    },
+
+    remove_rating_view: function() {
+        // Remove the rating view if it exists. Uses rating_view.close, so that the container element is not removed
```

Comment:
When this is called, the intent is that we're totally done with the view and want to get rid of it. The reference is deleted so that the object can be garbage collected. I want to clear the html of the view, but I also want to keep the view's `el` (which belongs to the template of the parent view) attached to the DOM, so that if we make a new RatingView later we can attach it to the same element with no new logic.